sql/postgresql/PostgreSQLParser.g4

```diff
@@ -0,0 +1,4323 @@
+               | constinterval opt_interval
+               | constinterval OPEN_PAREN iconst CLOSE_PAREN
+;
+ consttypename: numeric
+              | constbit
+              | constcharacter
+              | constdatetime
+;
+ generictype: type_function_name opt_type_modifiers
+            | type_function_name attrs opt_type_modifiers
+;
+ opt_type_modifiers: OPEN_PAREN expr_list CLOSE_PAREN
+                   |
+;
+ numeric: INT_P
+        | INTEGER
+        | SMALLINT
+        | BIGINT
+        | REAL
+        | FLOAT_P opt_float
+        | DOUBLE_P PRECISION
+        | DECIMAL_P opt_type_modifiers
+        | DEC opt_type_modifiers
+        | NUMERIC opt_type_modifiers
+        | BOOLEAN_P
+;
+ opt_float: OPEN_PAREN iconst CLOSE_PAREN
+          |
+;
+ bit: bitwithlength
+    | bitwithoutlength
+;
+ constbit: bitwithlength
+         | bitwithoutlength
+;
+ bitwithlength: BIT opt_varying OPEN_PAREN expr_list CLOSE_PAREN;
+
+ bitwithoutlength: BIT opt_varying;
+
+character: characterWithLength
+          | characterWithoutLength
+;
+ constcharacter: characterWithLength
+               | characterWithoutLength
+;
+ characterWithLength: character_c OPEN_PAREN iconst CLOSE_PAREN;
+
+ characterWithoutLength: character_c;
+
+ character_c: CHARACTER opt_varying
+          | CHAR_P opt_varying
+          | VARCHAR
+          | NATIONAL CHARACTER opt_varying
+          | NATIONAL CHAR_P opt_varying
+          | NCHAR opt_varying
+;
+ opt_varying: VARYING
+            |
+;
+ constdatetime: TIMESTAMP OPEN_PAREN iconst CLOSE_PAREN opt_timezone
+              | TIMESTAMP opt_timezone
+              | TIME OPEN_PAREN iconst CLOSE_PAREN opt_timezone
+              | TIME opt_timezone
+;
+ constinterval: INTERVAL;
+//TODO with_la was used
+ opt_timezone: WITH TIME ZONE
+             | WITHOUT TIME ZONE
+             |
+;
+ opt_interval: YEAR_P
+             | MONTH_P
+             | DAY_P
+             | HOUR_P
+             | MINUTE_P
+             | interval_second
+             | YEAR_P TO MONTH_P
+             | DAY_P TO HOUR_P
+             | DAY_P TO MINUTE_P
+             | DAY_P TO interval_second
+             | HOUR_P TO MINUTE_P
+             | HOUR_P TO interval_second
+             | MINUTE_P TO interval_second
+             |
+;
+ interval_second: SECOND_P
+                | SECOND_P OPEN_PAREN iconst CLOSE_PAREN
+;
+//precendence accroding to Table 4.2. Operator Precedence (highest to lowest)
+//https://www.postgresql.org/docs/12/sql-syntax-lexical.html#SQL-PRECEDENCE
+ a_expr: c_expr
+        //::	left	PostgreSQL-style typecast
+       | a_expr TYPECAST typename
+       | a_expr COLLATE any_name
+       | a_expr AT TIME ZONE a_expr
+
+       //right	unary plus, unary minus
+       | PLUS a_expr
+       | MINUS a_expr
+
+        //left	exponentiation
+       | a_expr CARET a_expr
+
+        //left	multiplication, division, modulo
+       | a_expr STAR a_expr
+       | a_expr SLASH a_expr
+       | a_expr PERCENT a_expr
```

Comment:
As I already wrote, it should be merged because they have the same priority correspondent to precedence from doc.

```suggestion
       | a_expr (STAR | SLASH | PERCENT) a_expr
```

It's related to other operators with the same priority as well.